tests/by-util/test_tar.rs

```diff
@@ -112,5 +217,478 @@ fn test_extract_nonexistent_archive() {
+        .arg("-xf")
+        .arg(at.plus("archive.tar"))
+        .current_dir(at.as_string())
+        .succeeds();
+
+    // Verify contents
+    assert_eq!(at.read("file-with-dash.txt"), "dash content");
+    assert_eq!(at.read("file_with_underscore.txt"), "underscore content");
+    assert_eq!(at.read("file.multiple.dots.txt"), "dots content");
+}
+
+// -----------------------------------------------------------------------------
+// 5. Error Handling and Exit Code Tests
+// -----------------------------------------------------------------------------
+
+#[test]
+#[cfg(unix)]
+fn test_create_permission_denied() {
+    use std::fs;
+    use std::os::unix::fs::PermissionsExt;
+
+    let (at, mut ucmd) = at_and_ucmd!();
+
+    at.write("file.txt", "content");
+    at.mkdir("readonly");
+
+    // Make directory read-only
+    let perms = fs::Permissions::from_mode(0o444);
+    fs::set_permissions(at.plus("readonly"), perms).unwrap();
+
+    ucmd.args(&["-cf", "readonly/archive.tar", "file.txt"])
+        .fails()
+        .code_is(2)
+        .stderr_contains("readonly/archive.tar");
+
+    // Cleanup - restore permissions so test cleanup can work
+    let perms = fs::Permissions::from_mode(0o755);
+    fs::set_permissions(at.plus("readonly"), perms).unwrap();
+}
+
+#[test]
+fn test_extract_corrupted_archive() {
+    let (at, mut ucmd) = at_and_ucmd!();
+
+    // Create a corrupted tar file (invalid header)
+    at.write("corrupted.tar", "This is not a valid tar file content");
+
+    ucmd.args(&["-xf", "corrupted.tar"]).fails().code_is(2);
+}
+
+#[test]
+fn test_create_with_dash_in_filename() {
+    let (at, mut ucmd) = at_and_ucmd!();
+
+    // Create files starting with dash
+    at.write("-dash-file.txt", "content with dash");
+    at.write("normal.txt", "normal content");
+
+    ucmd.args(&["-cf", "archive.tar", "--", "-dash-file.txt", "normal.txt"])
+        .succeeds();
+
+    assert!(at.file_exists("archive.tar"));
+
+    // Verify extraction works
+    at.remove("-dash-file.txt");
+    at.remove("normal.txt");
+
+    new_ucmd!()
+        .arg("-xf")
+        .arg(at.plus("archive.tar"))
+        .current_dir(at.as_string())
+        .succeeds();
+
+    assert!(at.file_exists("-dash-file.txt"));
+    assert_eq!(at.read("-dash-file.txt"), "content with dash");
+}
+
+// -----------------------------------------------------------------------------
+// 6. Verbose Output Format Tests
+// -----------------------------------------------------------------------------
+
+#[test]
+fn test_verbose_output_format_matches_gnu() {
+    let (at, mut ucmd) = at_and_ucmd!();
+
+    at.write("file1.txt", "content");
+    at.write("file2.txt", "content");
+
+    let result = ucmd
+        .args(&["-cvf", "archive.tar", "file1.txt", "file2.txt"])
+        .succeeds();
+
+    let stdout = result.stdout_str();
+
+    // Verify verbose output contains filenames
+    assert!(stdout.contains("file1.txt"));
+    assert!(stdout.contains("file2.txt"));
+}
```

Comment:
This test is more or less the same as `test_create_verbose`, with the only difference that this test uses two files.
```suggestion
```